src/Symfony/Component/Lock/Lock.php

```diff
@@ -32,22 +32,44 @@
     private $store;
     private $key;
     private $ttl;
+    private $autoRelease;
+    private $dirty = false;
 
     /**
-     * @param Key            $key
-     * @param StoreInterface $store
-     * @param float|null     $ttl
+     * @param Key            $key         Resource to lock
+     * @param StoreInterface $store       Store used to handle lock persistence
+     * @param float|null     $ttl         Maximum expected lock duration in seconds
+     * @param bool           $autoRelease Whether to automatically release the lock or not
      */
-    public function __construct(Key $key, StoreInterface $store, $ttl = null)
+    public function __construct(Key $key, StoreInterface $store, $ttl = null, $autoRelease = true)
     {
         $this->store = $store;
         $this->key = $key;
         $this->ttl = $ttl;
+        $this->autoRelease = (bool) $autoRelease;
 
         $this->logger = new NullLogger();
     }
 
     /**
+     * Automatically releases the underlying lock when the object is destructed.
+     */
+    public function __destruct()
+    {
+        if (!$this->autoRelease) {
+            return;
+        }
+
+        try {
```

Comment:
is the try catch actually required?

Reply:
to not throw an exception when the auto-release try to release the lock...

leave the try/cacth and you may miss exception when you count on the auto-release to release the lock
remove the try/catch and you may have an unexpected exception when you manually release the lock and the auto-release retry to do it

Reply:
isn't "release" idempotent?

Reply:
indeed. Fixed